plugins/flexibleengine/src/main/java/org/eclipse/xpanse/orchestrator/plugin/flexibleengine/monitor/MetricsService.java

```diff
@@ -199,6 +269,67 @@ public List<Metric> getMetrics(ResourceMetricRequest resourceMetricRequest) {
         return metrics;
     }
 
+    /**
+     * Get metrics of the @DeployService.
+     *
+     * @param serviceMetricRequest The request model to query metrics.
+     * @return Returns list of metric result.
+     */
+    public List<Metric> getMetricsForService(ServiceMetricRequest serviceMetricRequest) {
+        List<DeployResource> deployResources = serviceMetricRequest.getDeployResources();
+        CredentialDefinition credential = serviceMetricRequest.getCredential();
+        MonitorResourceType resourceType = serviceMetricRequest.getMonitorResourceType();
+        String region = deployResources.get(0).getProperties().get("region");
+        Project project = null;
+        if (StringUtils.isNotBlank(region)) {
+            String projectQueryUrl =
+                    flexibleEngineMonitorConverter.buildProjectQueryUrl(region).toString();
+            project =
+                    queryProjectInfo(credential, projectQueryUrl);
+        }
+        List<Metric> metrics = new ArrayList<>();
+        if (Objects.nonNull(project) && StringUtils.isNotBlank(project.getId())) {
+            Map<String, List<MetricInfoList>> deployResourceMetricInfoMap = new HashMap<>();
+            for (DeployResource deployResource : deployResources) {
+                List<MetricInfoList> targetMetricInfoList =
+                        getTargetMetricInfoList(deployResource, credential, resourceType, project);
+                deployResourceMetricInfoMap.put(deployResource.getResourceId(),
+                        targetMetricInfoList);
+            }
+            BatchListMetricDataRequest batchListMetricDataRequest =
+                    FlexibleEngineMonitorConverter.buildBatchListMetricDataRequest(
+                            serviceMetricRequest, deployResourceMetricInfoMap);
+
+            String url = FlexibleEngineMonitorConverter.getBatchQueryMetricBasicUrl(region,
+                    project.getId()).toString();
+
+            BatchListMetricDataResponse batchListMetricDataResponse =
+                    batchQueryListMetricsInfo(credential, batchListMetricDataRequest,
+                            url);
+
+            metrics = FlexibleEngineMonitorConverter.convertBatchListMetricDataResponseToMetric(
+                    batchListMetricDataResponse, deployResourceMetricInfoMap, deployResources);
+
+            if (!CollectionUtils.isEmpty(metrics)) {
```

Comment:
please move the cache steps to a private method. otherwise the method is simply too long.